server/database/dbOperations.ts

```diff
@@ -229,3 +239,397 @@ export const removeTableFromConfig = async (
+
+    // Create incident-specific data if provided
+    if (incidentData && collection.collection_type === "incident") {
+      const incidentQuery = `
+        INSERT INTO incidents (collection_id, incident_type, responsible_party, impact_description, supporting_evidence)
+        VALUES ($1, $2, $3, $4, $5)
+      `;
+
+      await pgClient.query(incidentQuery, [
+        newCollection.id,
+        incidentData.incident_type,
+        incidentData.responsible_party,
+        incidentData.impact_description,
+        JSON.stringify(incidentData.supporting_evidence || {}),
+      ]);
+    }
+
+    // Add entries if provided
+    if (entries && entries.length > 0) {
+      for (const entry of entries) {
+        // Fetch source data from the original table
+        const sourceDataQuery = `SELECT * FROM "${entry.source_table}" WHERE _id = $1 OR id = $1 LIMIT 1`;
+        const sourceResult = await pgClient.query(sourceDataQuery, [
+          entry.source_id,
+        ]);
+
+        const entryQuery = `
+          INSERT INTO collection_entries (collection_id, source_table, source_id, source_data, added_by, notes)
+          VALUES ($1, $2, $3, $4, $5, $6)
+        `;
+
+        await pgClient.query(entryQuery, [
+          newCollection.id,
+          entry.source_table,
+          entry.source_id,
+          JSON.stringify(sourceResult.rows[0] || {}),
+          collection.created_by,
+          entry.notes,
+        ]);
+      }
+    }
+
+    await pgClient.query("COMMIT");
+
+    return {
+      ...newCollection,
+      metadata: JSON.parse(newCollection.metadata || "{}"),
+    };
+  } catch (error) {
+    await pgClient.query("ROLLBACK");
+    throw error;
+  }
+};
+
+export const getAnnotatedCollection = async (
+  db: DatabaseConnection,
+  collectionId: string,
+): Promise<{
+  collection: AnnotatedCollection;
+  incident?: Incident;
+  entries: CollectionEntry[];
+}> => {
+  const pgClient = db as Client;
+
+  // Get the main collection
+  const collectionQuery = `SELECT * FROM annotated_collections WHERE id = $1`;
+  const collectionResult = await pgClient.query(collectionQuery, [
+    collectionId,
+  ]);
+
+  if (collectionResult.rows.length === 0) {
+    throw new Error("Collection not found");
+  }
+
+  const collection = {
+    ...collectionResult.rows[0],
+    metadata: JSON.parse(collectionResult.rows[0].metadata || "{}"),
+  };
+
+  // Get incident data if it's an incident collection
+  let incident = null;
+  if (collection.collection_type === "incident") {
+    const incidentQuery = `SELECT * FROM incidents WHERE collection_id = $1`;
+    const incidentResult = await pgClient.query(incidentQuery, [collectionId]);
+
+    if (incidentResult.rows.length > 0) {
+      incident = {
+        ...incidentResult.rows[0],
+        supporting_evidence: JSON.parse(
+          incidentResult.rows[0].supporting_evidence || "{}",
+        ),
+      };
+    }
+  }
+
+  // Get collection entries
```

Comment:
Whatever language is decided on for `createAnnotatedCollection`, let's follow suit here and in the fns below.